lib/iris/fileformats/grib/__init__.py

```diff
@@ -139,6 +139,7 @@ class GribWrapper(object):
     def __init__(self, grib_message, grib_fh=None):
         """Store the grib message and compute our extra keys."""
         self.grib_message = grib_message
+        self.data_dtype = np.zeros(0).dtype
```

Comment:
I moved it here because I need an attribute called `data_dtype` one this object. I don't think the change would cause any issue . I assume `np.zeros(0).dtype` will always give the same result.